poo/abstraction/en/Abstraction.md

```diff
@@ -0,0 +1,117 @@
+# Abstraction
+
+The **abstraction** idea, is chopper a class in parts, to be able to share (via a parameter passed in method for example) **an only** of this part rather the object “in full”.
+To understand the utility of this principle and how apply, take for example a class Human, since this is a thing that we know better.
+
+```java
+
+class Human {
+
+    int air;
+    int satiety;
+    float energy;
+
+
+    void breathe() {
+        air += 10;
+    }
+
+    void eat() {
+        satiety += 5;
+    }
+
+    void move() {
+        System.out.println("Un humain souhaite se déplacer !");
+    }
+
+    void consumeEnergy(float value) {
+        this.energie -= value;
+    }
+}
+```
+
+Now, imagine that we have a class Hiking which allows organise a walk enter Humans. A thing of style:
+
+```java
+class Hiking {
+
+    List<Human> participants;
+
+    public Hiking() { this.participants = new ArrayList<>(); }
+
+    void addParticipant(Human participant) { this.participants.add(participant); }
+
+    void moveAll() {
+        for(Human human : participants) {
+            human.move();
+            human.consumeEnergy(3);
+        }
+    }
+}
+```
+
+So far so good, we have that we want. This code works perfectly, except, in reality, he contain two major problems.
+
+1. **We restraint the hiking of the objects of type Human**, whereas anything can walk and exhausted could take a hiking. It’s not really the kind to forbid to elephants and to tractors to participate :(
+2. We ask a human in full but us all we need, this is his legs (hum). This class hiking, however currently access to the human respiratory system (via `breathe()` ) and to this stomach (via `eat()` ) ! **So, we give too much power to this class Hiking, since she is able to do things who shouldn’t be her accessible.**

```

Comment:
- Maybe `a whole human` is a better translation than `a human in full`.
- Also, reformulate `but us all we need, this is his legs` into `but all we need is his legs`. Maybe use `their` in replacement for `his` because we can't really use a gendered pronoun for an unknown human.
- Also `to this stomach` (`à cette estomac`) is incorrect. You meant to say `to his/her/their stomach`.
- Finally, a class is not a living person, so its pronoun is `it` (`she is able to` -> `it is able to`)
```suggestion
2. We ask a whole human but all we need is their legs (hum). This class hiking, however currently access to the human respiratory system (via `breathe()` ) and to their stomach (via `eat()` ) ! **So, we give too much power to this class Hiking, since it is able to do things who shouldn’t be accessible to it.**

```

Reply:
I applied the modifications,

>  a human in full
has been replaced to
>  a whole human
--
> but us all we need, this is his legs
by
> but all we need is his legs
--
Oups... a `t` slipped 😅
> to this stomach
has been replaced by
> to his stomach
--
> she is able to
has been replaced by
> it is able to
Thank for this details!

Reply:
Cool! _BTW, you should skip a line after quote to exit the quote block in Github_